### DISCUSSION TITLE
How to wrap Quasar elements which are not (yet) part of NiceGUI?

### DISCUSSION THREAD

**Discussion by rodja** [zauberzeug/nicegui#99] [Q&A]
@hroemer asked this question in #94.

**Comment by rodja** [comment]
You can simply create a new element in the same style as done internally by NiceGUI. For example the QLinearProgress from JustPy is wrapped like this:

```python
class linear_progress(nicegui.elements.float_element.FloatElement):

    def __init__(self, *, value: float = 0):
        view = jp.QLinearProgress(value=value, temp=False)
        super().__init__(view, value=value, on_change=None)
```

If you want to update the value, there is no need to access the underlying JustPy view for data bindings. Simply use the `set_value` method:

```python
progress = linear_progress()
ui.timer(0.01, lambda: progress.set_value(progress.value + 0.001))
```

Or use [NiceGUI bindings](https://nicegui.io/#bindings) if you have a data model (like a Downloader class which provides a float between 0 and 1 as `progress` property):

```python
linear_progress().bind_value_from(downloader, 'progress')
```

**Reply by hroemer** [reply]
Great example, by using `bind_value_from` I was able to get rid of `data` attribute requirement I had.
Thank you so much!

**Reply by hroemer** [reply]
I integrated your recommendations from #94 into the provided examples and opened a PR #101 for further discussion. I hope it makes sense at all.
